src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -86,4 +106,29 @@ private static string GetArchitecture()
 
         throw new OpenApiTaskFailedException("Unknown processor architecture encountered");
     }
+
+    private async Task GenerateSpectralReport(string spectralExecutePath, string swaggerDocumentPath, string rulesetUrl, string htmlReportPath, CancellationToken cancellationToken)
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            await this.AssignExecutePermission(spectralExecutePath, cancellationToken);
+        }
+
+        var exitCode = await this._processWrapper.RunProcessAsync(spectralExecutePath, new[] { "lint", swaggerDocumentPath, "--ruleset", rulesetUrl, "--format", "html", "--output.html", htmlReportPath }, cancellationToken);
+        if (exitCode != 0)
+        {
+            throw new OpenApiTaskFailedException($"Spectral report for {swaggerDocumentPath} could not be created.");
+        }
+
+        this._loggerWrapper.LogMessage("Spectral report generated. {0}", htmlReportPath);
+    }
+
+    private async Task AssignExecutePermission(string spectralExecutePath, CancellationToken cancellationToken)
+    {
+        var chmodExitCode = await this._processWrapper.RunProcessAsync("/bin/bash", new[] { "-c",  $"chmod +x {spectralExecutePath}" }, cancellationToken);
```

Comment:
I believe we could directly use `chmod` as the main input file/program?
Also I already experienced an issue where when the path contains a space, it fails.

Already dealt with this here: https://github.com/asimmon/ephemeral-mongo/blob/dda10a90f758e14447a74da0e6c451317e178a7f/src/EphemeralMongo.Core/FileSystem.cs#L29

Here because we use CliWrap all arguments (individual array items) are already escaped in case there's spaces and such

Reply:
I tried with the individual array items and it failed. https://github.com/gsoft-inc/wl-openapi-msbuild/pull/7/commits/6ace0613aebc82a4ecfd8e966892e3df94707142. Not sure why

Reply:
As for the whitespace in path, do we need to escape path when we dictate how this executable path is created?